src/AppBundle/Entity/Order.php

```diff
@@ -275,6 +278,10 @@ public function setRestaurant(Restaurant $restaurant)
     {
         $this->restaurant = $restaurant;
 
+        if (null !== $this->delivery) {
+            $this->delivery->setPriceFromOrder($this);
```

Comment:
Need to call `setOriginAddressFromOrder` as well?

Reply:
Done. Also added a unit test.